components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/WebhookManagementDAOImpl.java

```diff
@@ -202,15 +210,27 @@ public boolean isWebhookEndpointExists(String endpoint, int tenantId) throws Web
     }
 
     @Override
-    public void activateWebhook(String webhookId, int tenantId) throws WebhookMgtException {
+    public void activateWebhook(String webhookId, int tenantId, List<Subscription> channels, WebhookStatus status,
+                                String webhookEndpoint) throws WebhookMgtException {
 
-        updateWebhookStatus(webhookId, tenantId, WebhookStatus.ACTIVE.name());
+        processWebhookStatusUpdate(webhookId, tenantId, channels, status,
+                ERROR_CODE_WEBHOOK_STATUS_UPDATE_ERROR);
     }
 
     @Override
-    public void deactivateWebhook(String webhookId, int tenantId) throws WebhookMgtException {
+    public void deactivateWebhook(String webhookId, int tenantId, List<Subscription> channels, WebhookStatus status,
```

Comment:
same case here

Reply:
Refactored and improved the redability
fixed with https://github.com/wso2/carbon-identity-framework/pull/6885/commits/fb575c056228c93d2221b4ea29c500a76d09311c